apps/greenhouse-management/src/components/OrgInfo.jsx

```diff
@@ -1,23 +1,51 @@
-import React, { useMemo } from "react"
-import { Stack } from "juno-ui-components"
+import React, { useMemo, useEffect, useState } from "react"
+import { createClient } from "sapcc-k8sclient"
+import { useApiEndpoint, useAuthData } from "./StoreProvider"
+
+// Shows organization info
 
 const OrgInfo = () => {
-  // temporay fix to display org name until we have the k8sclient to fetch the metadata
+  const apiEndpoint = useApiEndpoint()
+  const [org, setOrg] = useState(null)
+
+  const authData = useAuthData()
+
   const orgName = useMemo(() => {
-    const currentUrl = new URL(window.location.href)
-    let match = currentUrl.host.match(/^(.+)\.dashboard\..+/)
-    return match ? match[1] : currentUrl.searchParams.get("org")
-  }, [])
+    if (!authData?.raw?.groups) return null
+    const orgString = authData?.raw?.groups.find(
+      (g) => g.indexOf("organization:") === 0
+    )
+    if (!orgString) return null
+    return orgString.split(":")[1]
+  }, [authData?.raw?.groups])
+
+  const client = useMemo(() => {
+    if (!apiEndpoint || !authData?.JWT) return null
+    return createClient({ apiEndpoint, token: authData?.JWT })
+  }, [apiEndpoint, authData?.JWT])
+
+  useEffect(() => {
+    if (!client || !orgName) return
+    // plugin configs
+    client
+      .get(`/apis/greenhouse.sap/v1alpha1/organizations/${orgName}`)
+      .then((res) => {
+        setOrg({
+          name: res?.spec?.displayName,
+          description: res?.spec?.description,
+        })
+      })
+  }, [client, orgName])
 
   return (
     <div className="org-info p-8 mb-8 bg-theme-background-lvl-0">
```

Comment:
We could also display a loading... here until we get the data back. What do you think?